library/image_streamer_deployment_plan_facts.py

```diff
@@ -92,15 +123,22 @@ def __init__(self):
 
     def execute_module(self):
         name = self.module.params.get("name")
-
+        options = self.module.params.get("options")
         ansible_facts = {}
-
         if name:
-            deployment_plans = self.i3s_client.deployment_plans.get_by("name", name)
+            ansible_facts['deployment_plans'] = self.i3s_client.deployment_plans.get_by("name", name)
+            if ansible_facts['deployment_plans'] and options == 'usedby':
+                deployment_plan = ansible_facts['deployment_plans'][0]
+                if deployment_plan:
```

Comment:
Is it required as you are checking  ansible_facts['deployment_plans']  in previous if condition.